### DISCUSSION TITLE
Remove __call__ method from OpticalSystem?

### DISCUSSION THREAD

**Discussion by LouisDesdoigts** [LouisDesdoigts/dLux#106] [Ideas]
Hey @Jordan-Dennis @benjaminpope, so as you know there is a slight annoyance when working with `optax` if you want to apply more complex optimisation schemes (multiple LR's, optimisation schedules) via the `optax.multi_transform` function where you have to wrap the equinox object in a list. If you're not sure what I'm talking about look in the [pixel level calibration tutorial](https://louisdesdoigts.github.io/dLux/notebooks/flatfield_calibration/) where we wrap the optical system object in a list -> `model = [opt_model]`. 

So I've figured out this can pretty easily prevented just by removing the `OpticalSystem.__call__()` method and simply renaming it `OpticalSystem.propagate()`. This allows us to work with the pure equinox object without any silly list business or having to generate psfs in loss functions like this `psf = model[0]()`. 

Now the obvious downside to this is that we lose the ability to simply call the object as a method of generating psfs, which is slightly annoying but I don't think this really matters tbh. I also this this will become less of an issue when we split the `OpticalSystem` object into `Optics`, `Scene`, and `Detector`.

Anyway I'm thinking of going ahead with this on main (I want to use this non-list wrapping syntax for the Instrumental Calibration paper), but you guys might have some thoughts. Personally I think simplifying the optimisation syntax is definitely good, and that losing the `.__call__()` capability is pretty whatever, as `.propagate()` is what is actually happening when we call this anyway. 

Let me know!

**Comment by benjaminpope** [comment]
Yikes, this doesn't seem very equinoctial. I don't think I really understand the problem here - do you have like, a code snippet illustrating the current and proposed behaviour?
